### DISCUSSION TITLE
User Identity

### DISCUSSION THREAD

**Discussion by ucodetech** [jasontaylordev/CleanArchitecture#1120] [General]
How do I override the default Users endpoint that the template comes with, so i can add more data to user registration?

**Comment by jasontaylordev** [comment]
> How do I override the default Users endpoint that the template comes with, so i can add more data to user registration?

The Identity endpoints are built using ASP.NET Core Identity API Authorization; https://learn.microsoft.com/en-us/aspnet/core/security/authentication/identity-api-authorization?view=aspnetcore-8.0. I don't think out-of-the-box customisation is supported at this stage.

You could create your own API endpoints based on the built-in endpoints. This is the source code for the built-in endpoints; https://github.com/dotnet/aspnetcore/blob/main/src/Identity/Core/src/IdentityApiEndpointRouteBuilderExtensions.cs. 

Create your own version of `ApiEndPointRouteBuilderExtensions.cs` and customise `src/Infrastructure/Identity/ApplicationUser.cs` to add more data to user registration.

**Comment by ucodetech** [comment]
Thank you so much for your help.

So it means I can create my own user Auth service and then the endpoints
customized to my needs yes?

Please here is my request, I started dot net middle of last year, I come
from php background and am really interested in using clean architecture
template I want to learn it so fast so I can start using it for projects,
please if there is any video you can recommend or a project you have done
implementing a customized Auth service  using clean architecture template
approach I would appreciate it to lay my on it.

Thanks

**Comment by jasontaylordev** [comment]
I'm sorry, but I can't offer you any recommended resources. My suggestion is to contact the ASP.NET Core team. They are better equipped to help you with this issue related to the ASP.NET Core Identity API feature.
